app/src/main/java/pl/gittobefit/WorkoutDisplay/fragments/DisplayReceivedTraining.java

```diff
@@ -107,231 +141,183 @@ public void onResume() {
-                String durationDisplay = "";
-                String scheduleTypeDisplay = "";
-                switch (UserTrainings.getInstance().getTraining(UserTrainings.getInstance().getTrainingArrayList().size() - 1).getTrainingForm().getTrainingType()) {
-                    case "SPLIT":
-                    case "FBW":
-                        durationDisplay = "Czas treningu: " + UserTrainings.getInstance().getTraining(UserTrainings.getInstance().getTrainingArrayList().size() - 1).getTrainingForm().getDaysCount() + " dni";
-                        if( UserTrainings.getInstance().getTraining(UserTrainings.getInstance().getTrainingArrayList().size() - 1).getTrainingForm().getScheduleType().equals("PER_DAY"))
-                        {
-                            scheduleTypeDisplay = "Forma treningu: na dzień";
-                        }
-                        else
-                        {
-                            scheduleTypeDisplay = "Forma treningu: powtarzalny";
-                        }
-                        for (int i = 0; i < UserTrainings.getInstance().getTraining(UserTrainings.getInstance().getTrainingArrayList().size() - 1).getTrainingForm().getDaysCount(); i++)
-                        {
-                            buttonArrayList.get(i).setVisibility(View.VISIBLE);
-                        }
-
-                        break;
-                    case "CARDIO":
-                    case "FITNESS":
-                        durationDisplay = "Czas treningu: " + UserTrainings.getInstance().getTraining(UserTrainings.getInstance().getTrainingArrayList().size() - 1).getTrainingForm().getDuration() + " minut";
-                        if( UserTrainings.getInstance().getTraining(UserTrainings.getInstance().getTrainingArrayList().size() - 1).getTrainingForm().getScheduleType().equals("SERIES"))
-                        {
-                            scheduleTypeDisplay = "Forma treningu: seriowy";
-                        }
-                        else
-                        {
-                            scheduleTypeDisplay = "Forma treningu: obwodowy";
-                        }
-                        day1Button.setVisibility(View.VISIBLE);
-                        break;
-                }
-
-                trainingType.setText(trainingTypeDisplay);
-                trainingForm.setText(scheduleTypeDisplay);
-                trainingDuration.setText(durationDisplay);
+                changePlusToMinus(buttonArrayList.get(i), R.drawable.ic_baseline_horizontal_rule);
+                linearLayoutArrayList.get(i).setVisibility(View.VISIBLE);
+                recyclerViewArrayList.get(i).setVisibility(View.VISIBLE);
+            }
+        }
 
-                for (int i = 0; i < UserTrainings.getInstance().getTraining(UserTrainings.getInstance().getTrainingArrayList().size() - 1).getPlanList().size(); i++)
+        final Observer<Integer> exerciseInfoObserver = new Observer<Integer>() {
```

Comment:
nie wiem czy to dobrze ze podczas zmiany jednego cwiczenia przeładowuja sie wszystkie listy